elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -308,34 +308,69 @@ public boolean matchesId(String checkId) {
      *
      * @param loadClass the load class
      * @param requestScope the request scope
+     * @param ids a list of object identifiers to optionally load.  Can be empty.
      * @return a filtered collection of resources loaded from the datastore.
      */
     public static Set<PersistentResource> loadRecords(
             Class<?> loadClass,
-            Optional<FilterExpression> filterExpression,
+            List<String> ids,
+            Optional<FilterExpression> filter,
             Optional<Sorting> sorting,
             Optional<Pagination> pagination,
             RequestScope requestScope) {
 
+        EntityDictionary dictionary = requestScope.getDictionary();
+        FilterExpression filterExpression;
+
         DataStoreTransaction tx = requestScope.getTransaction();
 
         if (shouldSkipCollection(loadClass, ReadPermission.class, requestScope)) {
-            return Collections.emptySet();
+            if (ids.isEmpty()) {
+                return Collections.emptySet();
+            } else {
+                throw new InvalidObjectIdentifierException(ids.toString(), dictionary.getJsonAliasFor(loadClass));
+            }
         }
 
-        EntityDictionary dictionary = requestScope.getDictionary();
         if (pagination.isPresent() && !pagination.get().isDefaultInstance()
                 && !CanPaginateVisitor.canPaginate(loadClass, dictionary, requestScope)) {
             throw new InvalidPredicateException(String.format("Cannot paginate %s",
                     dictionary.getJsonAliasFor(loadClass)));
         }
 
-        Iterable list;
-        list = tx.loadObjects(loadClass, filterExpression, sorting,
-                pagination.map(p -> p.evaluate(loadClass)), requestScope);
-        Set<PersistentResource> resources = new PersistentResourceSet(list, requestScope);
-        resources = filter(ReadPermission.class, resources);
-        return resources;
+        Set<PersistentResource> newResources = new LinkedHashSet<>();
+        if (!ids.isEmpty()) {
+            String typeAlias = dictionary.getJsonAliasFor(loadClass);
+            newResources = requestScope.getNewPersistentResources().stream()
+                        .filter(resource -> typeAlias.equals(resource.getType())
+                                && ids.contains(resource.getUUID().orElse("")))
+                        .collect(Collectors.toSet());
+            FilterExpression idExpression = buildIdFilterExpression(ids, loadClass, dictionary, requestScope);
+
+            // Combine filters if necessary
+            filterExpression = filter
+                    .map(fe -> (FilterExpression) new AndFilterExpression(idExpression, fe))
+                    .orElse(idExpression);
+        } else {
+            filterExpression = filter.orElse(null);
+        }
+
+        Set<PersistentResource> existingResources = filter(ReadPermission.class,
```

Comment:
General comment: I know we're going to push out the refactor to clean all of this up, but we probably want to get to this as soon as we can after getting elide 4.0 into good shape.